corehq/apps/reports/standard/cases/case_data.py

```diff
@@ -552,7 +568,267 @@ def close_case_view(request, domain, case_id):
+        return reverse(self.urlname, args=(self.domain, self.case_id))
+
+    @property
+    def page_context(self):
+        context = {
+            "main_case_id": self.case_id,
+            "main_xform_id": self.xform_id,
+        }
+        context.update(self.delete_dict)
+        return context
+
+    @location_safe
+    def get_cases_and_forms_for_deletion(self, request, domain, case_id, form_id=None):
+        """
+        Given a case object, recursively checks the case's related submission forms and for each form,
+        the related cases it has affected, and so on, until it goes through all related cases and their
+        related forms. This recursion is capped at the MAX_CASE_COUNT and MAX_SUBCASE_DEPTH values defined above.
+        Updates the delete_dict dictionary with the sorted lists.
+
+        :param case_id: The main case targeted for deletion. If the case deletion began from form
+        deletion, this is the first case returned while iterating through the form's case (create) updates.
+        :param form_id: The form_id of the form being targeted for deletion. This is only applicable if
+        a user is deleting a form that had previously created a case.
+        :return: The final, updated delete_dict, with additional entries if this is a form driven deletion.
+        """
+        case_instance = safely_get_case(request, domain, case_id, include_deleted=True)
+        try:
+            case_data = self.walk_through_case_forms(case_instance, subcase_count=0)
+        except TooManyCases:
+            if form_id:
+                messages.error(request, _("Deleting this form would delete too many related cases. "
+                                          "Please navigate to the form's Case Changes and delete some "
+                                          "of the cases it created before attempting to delete this form."))
+            else:
+                messages.error(request, _("Deleting this case would delete too many related cases. "
+                                          "Please delete some of this cases' subcases before attempting"
+                                          "to delete this case."))
+            return {'redirect': True}
+
+        case_data['affected_cases'] = [case for case in case_data['affected_cases']
+                                       if case.id not in case_data['case_delete_list']]
+
+        # Reorganizing display groups for form driven case deletion
+        if form_id:
+            updated_delete_cases = case_data['delete_cases']
+            for case in case_data['delete_cases']:
+                for form in case.delete_forms:
+                    if form.is_primary:
+                        self.form_delete_cases_display.append(case)
+                        updated_delete_cases.remove(case)
+                        break
+            case_data['delete_cases'] = updated_delete_cases
```

Comment:
This line is unnecessary since `updated_delete_cases` references the same object as `case_data['delete_cases']`.
```suggestion
```